### PR TITLE
yolo pipeline image shape override support

### PR DESCRIPTION
**Pull request by bfineran**
adds support for adding x,y shape override for yolo pipeline inputs. both dimensions must be divisible by all model strides

example for overriding to (416,320)
```python
>>> from deepsparse import Pipeline
>>> yolo_pipeline = Pipeline.create("yolo", image_size=(416, 320))
```

### CODE REVIEW

File: src/deepsparse/yolo/pipelines.py [hunk 1]
```diff
@@ -158,7 +191,7 @@ def process_inputs(self, inputs: YOLOInput) -> List[numpy.ndarray]:
 
             if isinstance(image, str):
                 image = cv2.imread(image)
-                image = cv2.resize(image, dsize=self.input_shape)
+                image = cv2.resize(image, dsize=tuple(reversed(self.image_size)))
```
Comment (rahul-tuli) [inline src/deepsparse/yolo/pipelines.py:194]: is this cause channel dim is last in cv2?

File: src/deepsparse/yolo/utils/utils.py [hunk 2]
```diff
@@ -326,6 +330,100 @@ def yolo_onnx_has_postprocessing(model_path: str) -> bool:
...
+    :param dim: dimension index of the tensor to get the shape of
+    :return: shape of the tensor at the given dimension
+    """
+    return tensor.type.tensor_type.shape.dim[dim].dim_value
+
+
+def set_tensor_dim_shape(tensor: onnx.TensorProto, dim: int, value: int):
+    """
+    Sets the shape of the tensor at the given dimension to the given value
+    :param tensor: ONNX tensor to modify the shape of
```
Comment (rahul-tuli) [inline src/deepsparse/yolo/utils/utils.py:420]: nit: blank line b/w description, params

**Review by rahul-tuli — DISMISSED** [review]

**Review by markurtz — DISMISSED** [review]
Looks good, just noting for the future it would be good to start getting all of our ONNX utilities and code migrated into SparseZoo so we can share the functionality across DeepSparse and SparseML code pathways

File: src/deepsparse/yolo/pipelines.py (hunk 1 above)
Comment (bfineran) [reply src/deepsparse/yolo/pipelines.py:194]: I don't think that's it, but this was required for shapes to line up

**Review by markurtz — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**